src/cascadia/QueryExtension/AzureLLMProvider.cpp

```diff
@@ -40,13 +40,22 @@ static constexpr std::wstring_view expectedHostSuffix{ L".openai.azure.com" };
 
 namespace winrt::Microsoft::Terminal::Query::Extension::implementation
 {
-    void AzureLLMProvider::SetAuthentication(const Windows::Foundation::Collections::ValueSet& authValues)
+    void AzureLLMProvider::SetAuthentication(const winrt::hstring& authValues)
     {
-        _azureEndpoint = unbox_value_or<hstring>(authValues.TryLookup(endpointString).try_as<IPropertyValue>(), L"");
-        _azureKey = unbox_value_or<hstring>(authValues.TryLookup(keyString).try_as<IPropertyValue>(), L"");
         _httpClient = winrt::Windows::Web::Http::HttpClient{};
         _httpClient.DefaultRequestHeaders().Accept().TryParseAdd(applicationJson);
-        _httpClient.DefaultRequestHeaders().Append(L"api-key", _azureKey);
+
+        if (!authValues.empty())
+        {
+            // Parse out the endpoint and key from the authValues string
+            WDJ::JsonObject authValuesObject{ WDJ::JsonObject::Parse(authValues) };
```

Comment:
will this throw when somebody has a stored key in the old format?

Reply:
Nope! For AzureOpenAI and OpenAI, `authValues` is always empty in this function so we won't get here. For GithubCopilot, the vault key changed so there is no old format.

Reply:
Theeennn why is there code in it that assumes an auth value can exist

Reply:
Wait ignore my previous comment (my bad for some reason I thought this was in `TerminalPage`)

However it still won't crash because the blob is sent here in the new format from `TerminalPage`, there is nowhere where we send the blob over in the old `IValueSet` format